pkg/validator/validator.go

```diff
@@ -88,6 +88,7 @@ func (s *Validator) Parse(document []byte) (schema.GroupVersionKind, *unstructur
 // Validate takes a parsed resource as input and validates it against
 // its schema.
 func (s *Validator) Validate(obj *unstructured.Unstructured) error {
+	obj = obj.DeepCopy()
```

Comment:
Can we change this to a shallow copy? I.e. just copy the keys to a new map.

Reply:
I'm not sure a shallow copy will work, chances are that nested props will be added/modified.

Reply:
I think `metadata.generation` is set to `1` for example.

Reply:
I've reviewed the preprocessing done on the object. Only metadata fields are possibly modified. Seems like it'd be a pain to only deep copy metadata. Can use `newObj["metadata"] = runtime.DeepCopyJSONValue(obj["metadata"])`

Reply:
Done